csharp/ql/src/semmle/code/csharp/ir/implementation/raw/IRFunction.qll

```diff
@@ -0,0 +1,102 @@
+private import internal.IRInternal
+import Instruction
+import csharp
+
+private newtype TIRFunction =
+  MkIRFunction(Callable callable) {
+    Construction::callableHasIR(callable)
+  }
+
+/**
+ * Represents the IR for a function.
+ */
+class IRFunction extends TIRFunction {
+  Callable callable;
+
+  IRFunction() {
+    this = MkIRFunction(callable)
+  }
+
+  final string toString() {
+    result = "IR: " + callable.toString()
+  }
+
+  /**
+   * Gets the function whose IR is represented.
+   */
+  final Callable getCallable() {
```

Comment:
In all the places where you replaced the type `Function` with `Callable`, you also changed the name of the predicate to use `Callable`. If the predicate is to be shared between languages, it needs to have the same name. I'd recommend just changing the predicate names back to use `Function` for now. It's possible that some of the predicates that use the `Function` type don't actually need to be in the shared part of the IR, but I'd rather try to figure out how to factor them out later.